File: 230-kth-smallest-element-in-a-bst/kth-smallest-element-in-a-bst.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
int kthSmallest(struct TreeNode* root, int k) {
    int count = k;
    int a;
    void helper(struct TreeNode * root) {
        if (!root || !count)
            return;
        helper(root->left);
        if (count) {
            a = root->val;
            count--;
            helper(root->right);
        }
    }
    helper(root);
    return a;
}

File: 230-kth-smallest-element-in-a-bst/kth-smallest-element-in-a-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def helper(self, root: Optional[TreeNode]):
        if not root or not self.k:
            return
        self.helper(root.left)
        if self.k:
            self.a = root.val
            self.k -= 1
            self.helper(root.right)

    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        self.k = k
        self.helper(root)
        return self.a
